rubix-spi/src/main/java/com/qubole/rubix/spi/BookKeeperFactory.java

```diff
@@ -20,13 +24,84 @@
 import org.apache.thrift.transport.TTransport;
 import org.apache.thrift.transport.TTransportException;
 
+import java.net.SocketException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
 /**
  * Created by sakshia on 5/10/16.
  */
 public class BookKeeperFactory
 {
   BookKeeperService.Iface bookKeeper;
   private static Log log = LogFactory.getLog(BookKeeperFactory.class.getName());
+  private static final AtomicInteger count = new AtomicInteger();
+  private static final AtomicBoolean flag = new AtomicBoolean();
+  private static PoolConfig poolConfig;
+  private static ConcurrentHashMap<String, Integer> concurrentHashMap = new ConcurrentHashMap<>();
+  private static ObjectFactory<TSocket> factory;
+  static ObjectPool pool;
+
+  private static final String LOCALHOST = "localhost";
+
+  static {
+    // TODO: add config.properties file to set this config
+    poolConfig = new PoolConfig();
+    poolConfig.setPartitionSize(10);
```

Comment:
partition size limits the number of hosts we can have in a cluster. We shouldnt have this limit, instead grow the partitions in ObjectPool dynamically during registerHost call if index is out of bounds

Reply:
added logic of keeping Map<Host, Object Pool>, this way we can directly deal with the host and remove partition index logic.